src/OpenLoco/GameCommands/ChangeCompanyFace.cpp

```diff
@@ -0,0 +1,87 @@
+#include "../CompanyManager.h"
+#include "../Interop/Interop.hpp"
+#include "../Localisation/StringIds.h"
+#include "../Objects/CompetitorObject.h"
+#include "../Objects/ObjectManager.h"
+#include "../Ui/WindowManager.h"
+#include "GameCommands.h"
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x00435506
+    static uint32_t changeCompanyFace(uint8_t flags, CompanyId targetCompanyId, ObjectHeader& targetHeader)
+    {
+        GameCommands::setExpenditureType(ExpenditureType::Miscellaneous);
+
+        // Find target competitor object among loaded objects
+        auto foundCompetitor = ObjectManager::findObjectHandle(targetHeader);
+
+        // See whether any other company is using this competitor already
+        for (auto& company : CompanyManager::companies())
+        {
+            if (company.competitorId == foundCompetitor->id)
+            {
+                if (company.id() != targetCompanyId)
+                {
+                    GameCommands::setErrorText(StringIds::already_selected_for_another_company);
+                    return GameCommands::FAILURE;
+                }
+                else
+                {
+                    // No change; no work required
+                    return 0;
+                }
+            }
+        }
+
+        // Stop here if we're just querying
+        if ((flags & GameCommands::Flags::apply) == 0)
+            return 0;
+
+        // Do we need to load the competitor, still?
+        if (!foundCompetitor)
+        {
+            // Load the new competitor object
+            if (!ObjectManager::load(targetHeader))
+                return GameCommands::FAILURE;
+
+            ObjectManager::reloadAll();
+            Ui::WindowManager::close(Ui::WindowType::dropdown);
+
+            foundCompetitor = ObjectManager::findObjectHandle(targetHeader);
+            if (!foundCompetitor)
+                return GameCommands::FAILURE;
+        }
+
+        // Unload the target company's current competitor
+        auto* targetCompany = CompanyManager::get(targetCompanyId);
```

Comment:
Previously it double checked that the competitor id was not being used by any other company first before unloading.

Reply:
I noticed that, too, but didn't see the point as we already check this at the start. I suppose it would be possible to hack a save so two companies use the same competitor, though. If we want to guard against that, I'd rather adjust the first check to tally rather than breaking on the first match, though. Any thoughts on this?

Reply:
yes i think its sensible to guard it. Can combine it with the first loop

Reply:
I ended going for a separate loop, after all. Thoughts?